src/linux/keyboard_state.rs

```diff
@@ -25,16 +25,12 @@ impl Drop for KeyboardState {
 }
 
 impl KeyboardState {
-    pub fn new() -> Result<KeyboardState, ()> {
+    pub fn new() -> Option<KeyboardState> {
         unsafe {
             let dpy = xlib::XOpenDisplay(null());
-            if dpy.is_null() {
-                return Err(());
-            }
+            NonNull::new(dpy)?; //return None if dpy is null
```

Comment:
Are you sure about that ?

Reading rust's doc 

> If you're not sure if you should use NonNull<T>, just use *mut T!

I'm not really sure about NonNull use here...

Reply:
Yeah, this is a clever one-liner. The NonNull is never used. I'll change this back.